technical-docs/designs/infra/managed-identity.md

```diff
@@ -0,0 +1,251 @@
+# MSI Support Testing for Bedrock AKS-gitops
+
+| Revision | Date         | Author         | Remarks       |
+| -------: | ------------ | -------------- | ------------- |
+|      0.1 | Mar-30, 2020 | Nathaniel Rose | Initial Draft |
+
+## 1. Overview
+
+Managed Identities for Azure resources provides Azure services with an
+automatically managed identity in Azure AD. You can use the identity to
+authenticate to any service that supports Azure AD authentication, including Key
+Vault, without any credentials in your code. Terraform can be configured to use
+managed identity for authentication in one of two ways: using environment
+variables, or by defining the fields within the provider block.
+
+AKS creates two managed identities:
+
+- System-assigned managed identity: The identity that the Kubernetes cloud
+  provider uses to create Azure resources on behalf of the user.
+
+- User-assigned managed identity: The identity that's used for authorization in
+  the cluster.
+
+This document outlines a testing suite to support feature related support for
+managed identities in AKS using a proposed new Bedrock environment that
+leverages a modified cobalt project test harness in order for test pod identity
+within an AKS cluster using agile CI/CD and test validation.
+
+### Issues Addressed:
+
+1. [As an SRE, I want Enable MSI Support for aks-gitops module](https://github.com/microsoft/bedrock/issues/994)
+2. [As an Operator, I want automated testing validation for MSI verified within Bedrock](https://github.com/microsoft/bedrock/issues/1197)
+3. [As an operator, I want integration Tests tracking with junit logs from terratest](https://github.com/microsoft/bedrock/issues/867)
+4. [As an operator, I want to implement a managed service identity (via AAD Pod Identity) based secret handling strategy](https://github.com/microsoft/bedrock/issues/482)
+
+## 2. Out of Scope
+
+An existing pull request for Bedrock currently exists that enables MSI support
+for aks-gitops modules [#995](https://github.com/microsoft/bedrock/pull/995).
+This design document seeks to solely capture a terraform template and
+complementary test.
+
+The following are not included in this proposal:
+
+- Mocking for Terraform Unit Tests
+- Feature revert and Rollback from failed merges
+- Adjusting Cobalt Test Fixture support for current file organization of
+  Bedrock: i.e.: testing files in respective folders for template environments.
+
+## 3. Design Details
+
+This design seeks to introduce modular testing for terraform known as
+`Test Fixtures` based on best practices initially introduced by
+[Project Cobalt](github.com/microsoft/cobalt). The test fixtures decouples
+terraform commands to respective pipeline templats to be called and dynamically
+populated by a targeted template test.
+
+### 3.1 Embed new Infrastructure DevOps Model Flow - Continuous Integration
+
+Bedrock infrastructure integration tests have problematic gaps that do not
```

Comment:
Can you provide more details on the problematic gaps this new flow addresses.

Reply:
Updated, can you take another look to see if this clarifies.